aea/helpers/search/models.py

```diff
@@ -613,6 +1052,7 @@ def __init__(self, constraints: List[ConstraintExpr]):
         :param constraints: the list of constraints expression (in conjunction).
         """
         self.constraints = constraints
+        self.check_validity()
```

Comment:
I noticed `check_validity` of `And`, `Or`, `Not` are not called at all, so I added it 
(probably an oversight as they are called in the other repo which seems to have been more developed).

Reply:
Yes sounds good